src/lib.rs

```diff
@@ -587,11 +585,7 @@ mod tests {
 
         // Individual location line, e.g:
         //  "#1: Register #R0, size: 8"
-        let mut locs = Vec::with_capacity(num_locs as usize);
-        for _ in 0..num_locs {
-            let line = lines.next().unwrap();
-            locs.push(parse_loc(line));
-        }
+        let locs = lines.take(num_locs as usize).map(|x| parse_loc(x)).collect();
```

Comment:
Why type does `num_locs` have? Because `as` is so dangerous in Rust, I normally add a comment to each one saying why it's a safe things to do.

Reply:
`num_locs` is a `u16`, so we're okay here. I'll add a comment for clarity though.